javascript/javascript2/week1/danish.js

```diff
@@ -0,0 +1,27 @@
+function findNumberOfDanishLetters(str) {
+  let letterFrequency = {};
+  let totalOfLetters = 0;
+  const danishLetters = ["æ", "ø", "å"];
+  str = str.toLowerCase();
+  if (typeof str !== "string") {
+    return "Please input a string";
+  }
+  const words = str.split(" ");
+  words.forEach((str) => {
+    for (let char of str) {
+      if (danishLetters.includes(char)) {
+        if (!letterFrequency[char]) {
+          letterFrequency[char] = 1;
+        } else {
+          letterFrequency[char] = letterFrequency[char] + 1;
+        }
+        totalOfLetters++;
+      }
+    }
+  });
+  return { total: totalOfLetters, ...letterFrequency };
```

Comment:
You could store frequencies and the total in a single object instead of creating two instances (`letterFrequency` plus the one you return).
It is a small optimisation but it can have an impact on large projects.